### PR TITLE
feat: Vue3, Vue Router, Vuex, and Typescript Template

### PR DESCRIPTION
**Pull request by kmuchmore**
Adds a Vue3 Typescript template with Vuex and Vue Router.

### CODE REVIEW

**Comment by leaanthony** [comment]
I think I need to put a bullet in the hound.

**Comment by leaanthony** [comment]
Sorry for the slow response on this. What platforms have you tested this on?

File: cmd/templates/vue3-full/frontend/src/views/Home.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+export default defineComponent({
+  name: "Home",
+  components: {
+    HelloWorld,
+  },
+  setup() {
+    
+    const message = ref("Click the Icon");
+
+    const getMessage = () => {
```
Comment (leaanthony) [inline cmd/templates/vue3-full/frontend/src/views/Home.vue:31]: Without a polyfill, arrow functions will cause the windows builds to fail.

**Comment by leaanthony** [comment]
Can confirm this doesn't work on windows:



Digging into the docs and issue tracker, it seems that IE11 support is not available which puts me in a predicament. I don't really want to be fragmenting the templates but I don't want to hold V1 back whilst V2 is being developed. Before merging this I may need to only present templates that are available for a given platform... This would need to be handled by both `init` and `build`.

**Comment by leaanthony** [comment]
I've just merged #528 so that we can get this through. Could you please rebase your PR on develop and update your template to use the `Platforms` key to indicate the supported platforms. The best way of testing is set it to be `["windows"]` then see if `init` and `build` throw warnings. Thanks 👍

**Comment by leaanthony** [comment]
@kmuchmore - any updates? Cheers 👍

**Comment by kmuchmore** [comment]
I won’t have time until probably this weekend to look at this. Sorry

**Comment by leaanthony** [comment]
No rush and thanks for this awesome contribution!

**Comment by leaanthony** [comment]
I'm going to merge this as-is then update it to set the verified platforms.

**Pull request merged**